okhttp-sse/src/main/kotlin/okhttp3/sse/EventSource.kt

```diff
@@ -37,5 +40,40 @@ interface EventSource {
       request: Request,
       listener: EventSourceListener,
     ): EventSource
+
+    companion object {
+      /**
+       * Wraps a [Call.Factory] into [EventSource.Factory].
+       */
+      @JvmStatic
+      @JvmName("create")
+      fun Call.Factory.asEventSourceFactory(): Factory =
+        Factory { request, listener ->
+          val actualRequest =
+            if (request.header("Accept") == null) {
+              request.newBuilder().addHeader("Accept", "text/event-stream").build()
+            } else {
+              request
+            }
+
+          this.newCall(actualRequest).toEventSource(listener)
+        }
+    }
+  }
+
+  companion object {
+    /**
+     * Creates a new [EventSource] from the [Call] and immediately enqueue it.
+     */
+    @JvmStatic
+    @JvmName("create")
+    fun Call.toEventSource(listener: EventSourceListener): EventSource = RealEventSource(this, listener).also(this::enqueue)
```

Comment:
I don't like this name, as it hides the fact that it fires off the request using enqueue. I would want some kind of verb that coveys this in the function name

Reply:
Our existing naming would be `enqueueEventSource()`.